core/src/main/java/com/linecorp/armeria/server/RoutingTrie.java

```diff
@@ -261,8 +259,19 @@ private void dump(PrintWriter p, Node<V> node, int depth) {
          * @param value the value belonging to the path
          */
         Builder<V> add(String path, V value) {
+            return add(path, value, true);
+        }
+
+        /**
+         * Adds a path and a value to be built as {@link RoutingTrie}.
+         *
+         * @param path  the path to serve
+         * @param value the value belonging to the path
+         * @param hasHighPrecedence whether the path being added has high precedence or not
```

Comment:
Would be good to document what the precedence does. I'm not totally sure but is it `Any values that match a path with high precedence will be returned in front of matches with low precedence. Within the same precedence, values are returned in the order the paths were defined.`

Reply:
Updated as advised.